### PR TITLE
Messaging Content Card Support

### PR DESCRIPTION
**Pull request by dsoffiantini**
## Description

Adds support for content cards and other proposition types for react native

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by cacheung** [comment]
Any documentation update?  Or it will be in another PR?

File: apps/AEPSampleApp/ios/Podfile.lock [hunk 1]
```diff
@@ -161,12 +161,10 @@ PODS:
   - RCTAEPEdgeIdentity (6.0.2):
     - AEPEdgeIdentity (< 6.0.0, >= 5.0.0)
     - React
-  - RCTAEPMessaging (6.0.2):
-    - AEPMessaging (< 6.0.0, >= 5.0.0)
-    - glog
-    - RCT-Folly (= 2022.05.16.00)
-    - React-Core
-  - RCTAEPOptimize (6.0.2):
+  - RCTAEPMessaging (6.0.3):
```
Comment (cacheung) [inline apps/AEPSampleApp/ios/Podfile.lock:164]: Do you want to build the sample app with 6.0.4 version?

File: packages/messaging/src/models/ContentCard.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ANY KIND, either express or implied. See the License for the specific
+    language governing permissions and limitations under the License.
+*/
+
+import { PersonalizationSchema } from './PersonalizationSchema';
+
+type ContentCardTemplate = 'SmallImage';
+type DismissButtonStyle = 'circle' | 'none' | 'simple';
+
+export interface ContentCard {
```
Comment (cacheung) [inline packages/messaging/src/models/ContentCard.ts:18]: Where do we use this interface?
Is this for UI?  
Can we update the sample app to show the user how to implement with the UI?

File: apps/AEPSampleApp/ios/Podfile.lock (hunk 1 above)
Comment (dsoffiantini) [reply apps/AEPSampleApp/ios/Podfile.lock:164]: I can upgrade it if youd like, that one came out after I had worked on this and is fresh off the press

File: packages/messaging/src/models/ContentCard.ts (hunk 2 above)
Comment (dsoffiantini) [reply packages/messaging/src/models/ContentCard.ts:18]: We have a different UI project that may come later for them. ContentCard is part of a union type for Proposition Item so they can get it from the `getProposition` call

Comment (cacheung) [reply packages/messaging/src/models/ContentCard.ts:18]: We think it is still good to show how to retrieve this in the sample app in a log window.  Possible to implement this in the sample app?

And any documentation should be added for this ability for the release?

File: apps/AEPSampleApp/ios/Podfile.lock (hunk 1 above)
Comment (cacheung) [reply apps/AEPSampleApp/ios/Podfile.lock:164]: It is fine to leave it as is.

**Review by cacheung — APPROVED** [review]

**Pull request merged**